clippy_lints/src/mixed_read_write_in_expression.rs

```diff
@@ -136,6 +137,21 @@ impl<'a, 'tcx> DivergenceVisitor<'a, 'tcx> {
 impl<'a, 'tcx> Visitor<'tcx> for DivergenceVisitor<'a, 'tcx> {
     fn visit_expr(&mut self, e: &'tcx Expr<'_>) {
         match e.kind {
+            // fix #10776
+            ExprKind::Block(block, ..) => {
+                if let Some(e) = block.expr {
+                    self.visit_expr(e);
+
+                    return;
+                }
+
+                if let Some(stmt) = block.stmts.first() && block.stmts.len() == 1 {
```

Comment:
Should be `if let [stmt] = block.stmts`. Everything else looks good.

Reply:
Good point, forgot you could do that

Reply:
Fixed